src/game/objects/puzzle_hole.c

```diff
@@ -45,7 +45,7 @@ void PuzzleHoleCollision(
             return;
         }
 
-        if (lara_item->frame_number == AF_USEPUZZLE) {
+        if (lara_item->frame_number == (AF_USEPUZZLE * ANIM_SCALE)) {
```

Comment:
Unnecessary braces (I will stop with this comment here, please go through the diff and apply to all cases)